src/pkg/packager/publish.go

```diff
@@ -0,0 +1,314 @@
+	case ".txt":
+		return ZarfLayerMediaTypeTxt
+	default:
+		return ZarfLayerMediaTypeUnknown
+	}
+}
+
+// Publish publishes the package to a registry
+//
+// This is a wrapper around the oras library
+// and much of the code was adapted from the oras CLI - https://github.com/oras-project/oras/blob/main/cmd/oras/push.go
+//
+// Authentication is handled via the Docker config file created w/ `zarf tools registry login`
+func (p *Packager) Publish() error {
+	p.cfg.DeployOpts.PackagePath = p.cfg.PublishOpts.PackagePath
+	if err := p.loadZarfPkg(); err != nil {
+		return fmt.Errorf("unable to load the package: %w", err)
+	}
+
+	paths := []string{
+		p.tmp.ZarfYaml,
+		filepath.Join(p.tmp.Base, "sboms.tar.zst"),
+	}
+	componentDirs, err := filepath.Glob(filepath.Join(p.tmp.Base, "components", "*"))
+	if err != nil {
+		return err
+	}
+	componentTarballs := []string{}
+	// repackage the component directories into tarballs
+	for _, componentDir := range componentDirs {
+		all, err := filepath.Glob(filepath.Join(componentDir, "*"))
+		if err != nil {
+			return err
+		}
+		dst := filepath.Join(p.tmp.Base, "components", filepath.Base(componentDir)+".tar.zst")
+		err = archiver.Archive(all, dst)
+		if err != nil {
+			return err
+		}
+		componentTarballs = append(componentTarballs, dst)
+	}
+	paths = append(paths, componentTarballs...)
+	imagesLayers, err := filepath.Glob(filepath.Join(p.tmp.Base, "images", "*"))
+	if err != nil {
+		return err
+	}
+	paths = append(paths, imagesLayers...)
+	ref, err := p.ref("")
+	if err != nil {
+		return err
+	}
+	message.HeaderInfof("📦 PACKAGE PUBLISH %s:%s", p.cfg.Pkg.Metadata.Name, ref.Reference)
+	err = p.publish(ref, paths)
+	if err != nil {
+		return fmt.Errorf("unable to publish package %s: %w", ref, err)
+	}
+	return nil
+}
+
+func (p *Packager) generateManifestConfigFile() (ocispec.Descriptor, []byte, error) {
+	// Unless specified, an empty manifest config will be used: `{}`
+	// which causes an error on Google Artifact Registry
+	// to negate this, we create a simple manifest config with some build metadata
+	// the contents of this file are not used by Zarf
+	type OCIConfigPartial struct {
+		Architecture string            `json:"architecture"`
+		OCIVersion   string            `json:"ociVersion"`
+		Annotations  map[string]string `json:"annotations,omitempty"`
+	}
+
+	annotations := map[string]string{
+		"org.opencontainers.image.title":       p.cfg.Pkg.Metadata.Name,
+		"org.opencontainers.image.description": p.cfg.Pkg.Metadata.Description,
+	}
+
+	manifestConfig := OCIConfigPartial{
+		Architecture: p.cfg.Pkg.Build.Architecture,
+		OCIVersion:   "1.0.1",
+		Annotations:  annotations,
+	}
+	manifestConfigBytes, err := json.Marshal(manifestConfig)
+	if err != nil {
+		return ocispec.Descriptor{}, nil, err
+	}
+	manifestConfigDesc := content.NewDescriptorFromBytes("application/vnd.unknown.config.v1+json", manifestConfigBytes)
+	return manifestConfigDesc, manifestConfigBytes, nil
+}
```

Comment:
Is there more the team would like to see stored in this manifest config? It is not used or consumed by Zarf at all, but it could be useful for storing some metadata we do not currently track in `zarf.yaml`.